File: Java/1464-Maximum-Product-of-Two-Elements-in-an-Array/soln.java
class Solution {
    public int maxProduct(int[] nums) {
        final int n = nums.length;
        int ans = 0;
        for(int i = 0; i < n; ++i) {
            for(int j = i + 1; j < n; ++j) {
                ans = Math.max(ans, (nums[i] - 1) * (nums[j] - 1));
            }
        }
        return ans;
    }
}

File: C++/0135-Candy/soln.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        int n = ratings.size();
        vector<int> candys(n, 1);
        for(int i = 1; i < n; ++i) {
            if (ratings[i] > ratings[i - 1]) candys[i] = candys[i - 1] + 1;
        }
        for(int i = n - 2; i >= 0; --i) {
            if (ratings[i] > ratings[i + 1]) candys[i] = max(candys[i], candys[i + 1] + 1);
        }
        int ans = 0;
        for(int candy : candys) ans += candy;
        return ans;
    }
};